spring-integration-event/src/main/java/org/springframework/integration/event/inbound/ApplicationEventListeningMessageProducer.java

```diff
@@ -116,20 +129,45 @@ public void onApplicationEvent(ApplicationEvent event) {
 		}
 	}
 
+	@Override
+	protected Object evaluatePayloadExpression(Object rootObject) {
+		if (!this.payloadExpressionSet && rootObject instanceof PayloadApplicationEvent) {
+			return ((PayloadApplicationEvent) rootObject).getPayload();
```

Comment:
We have an event inbound channel adapter that emits events as `Message<?>` where the payload can be the event or the result of evaluating an expression against the event.

This change by Artem simply unwraps `PayloadApplicationEvent` - that's been the debate all along - should the `?` be the wrapper or the original `Object` event passed to the publisher.

My personal feeling is that it should be the event by default (so the adapter always emits `Message<ApplicationEvent>` but I can see the other side (that the wrapper is an internal vehicle and should not be exposed).